### DISCUSSION TITLE
Security Implications - Frigate was pointed at public MQTT Broker for a year

### DISCUSSION THREAD

**Discussion by Gobbless** [blakeblackshear/frigate#9112] [Ask A Question]
Hello,

When I was first messing around with Frigate, I was very new to the home automation and IOT world. During my initial config setup, I slapped a public MQTT broker in the config just to get things running as I thought it was required to launch. In the last year, I've added cameras and started running Frigate full time. In all this time, I never removed the temporary public MQTT broker from my config file. I didn't use MQTT and didn't think about it at all.

I'm wondering if there are any security ramifications of this? I am still very new to MQTT and my understanding is simplistic. Does Frigate receive any commands via MQTT? Or Just send? Are any of these MQTT messages Frigate sends sensitive data? Is there any way I could've exposed my internet connected machine with unrestricted traffic out (I know) to anything by talking to this public broker?

Basically, I'm wondering if there's any reason I should nuke my instance and rebuild or if I can say with relative certainty that changing my config to remove the broker is good enough.

Thanks everyone.

**Comment by NickM-27** [comment]
frigate can receive predefined commands like enabling or disabling recordings, detection, etc. but not arbitrary commands. It exposes thumbnails (crops of the objects that are detected) but no recordings or anything like that directly

**Reply by Gobbless** [reply]
I checked out the docs again and appreciate your reply for clarification as well. I am wondering if the MQTT commands are reset when the frigate instance is restarted or if they are saved? Things like improve_contrast, motion_threshold, and motion_contour_area specifically. Detect, recordings, snapshots, and motion are all simple enough to check from the UI in terms of whether they are on or off. Well, snapshots is a little confusing as well because I'm not sure if its referencing the actual snapshots in general during event captures or just the snapshot sending over MQTT feature. Thanks again.

**Reply by NickM-27** [reply]
in frigate 0.12 they get reset on restart

**Reply by Gobbless** [reply]
0.12 and forward or just 0.12? I'm on 0.13 beta for the Frigate+ testing. If it's 0.12 and forward, would just a restart should reset them? If they had been set before 0.12 on an older version, would the update to 0.12 reset them as well? Basically, could those values have been set and stored in the pre 0.12 version and then referenced as the new default in 0.12 and up? Meaning restarts of 0.12 and up would just reset them to their previously stored pre- 0.12 values that were set via MQTT? Or would they be reset to default? Sorry if this question is confusing - having trouble wording it.

**Reply by NickM-27** [reply]
starting in 0.12 and also in 0.13, restarts will not retain the values set via MQTT.

**Reply by Gobbless** [reply]
Even if those values were set and retained in a version pre 0.12? They would be "thrown out" and set to defaults once frigate was updated to version 0.12?

**Reply by NickM-27** [reply]
it sort of depends, it should be cleared out but there are some cases where it isn't and mqtt needs to be restarted to clear it out

**Reply by Gobbless** [reply]
By MQTT do you mean the MQTT broker server? Or is there an MQTT service inside Frigate itself that needs to be restarted? Just want to make sure I go through all the correct steps to get rid of this potentially bad data/settings.

**Reply by NickM-27** [reply]
mqtt broker

**Reply by Gobbless** [reply]
Ok perfect. I have completely changed MQTT brokers in my config while running version 0.13. It sounds like I am good at this point then considering the new broker couldn't have any of the old broker's stored messages.
